File: notes.pl
# perl notes
sub greet {
    my ($name) = @_;
    return "hello $name";
}

print greet("practice"), "\n";

my @nums = (1, 2, 3);
my @doubled = map { $_ * 2 } @nums;
print "@doubled\n";

# TODO: try references next

File: notes.kt
// kotlin notes
fun greet(name: String): String {
    return "hello $name"
}

fun main() {
    println(greet("practice"))

    val nums = listOf(1, 2, 3)
    val doubled = nums.map { it * 2 }
    println(doubled)
}
// TODO: try coroutines next
